src/node_lkl.cc

```diff
@@ -150,3 +148,39 @@ NAN_METHOD(syscall) {
 		}
 	}
 }
+
+NAN_METHOD(parseDirent64) {
```

Comment:
You're right that this is not a public API but internal APIs should also be usable and follow good practices. In this case, making this function return a new array to the caller sounds like a much more idiomatic JS interface.

Regarding performance, I think the overhead will be too low and also we can amortise it by making the buffer hold something like 100 entries so that most `readdir` calls do only one iteration